windows/vNext/ReactNativeVideoCPP/ReactVideoView.cpp

```diff
@@ -0,0 +1,255 @@
+                  WriteProperty(eventDataWriter, L"orientation", orientation);
+                  eventDataWriter.WriteObjectEnd();
+                }
+
+                WriteProperty(eventDataWriter, L"canPlayFastForward", false);
+                WriteProperty(eventDataWriter, L"canPlaySlowForward", false);
+                WriteProperty(eventDataWriter, L"canPlaySlow", false);
+                WriteProperty(eventDataWriter, L"canStepBackward", false);
+                WriteProperty(eventDataWriter, L"canStepForward", false);
+              }
+              eventDataWriter.WriteObjectEnd();
+            });
+      }
+    }
+  });
+}
+
+void ReactVideoView::OnMediaFailed(IInspectable const &, IInspectable const &) {}
+
+void ReactVideoView::OnMediaEnded(IInspectable const &, IInspectable const &) {
+  runOnQueue([weak_this{get_weak()}]() {
+    if (auto strong_this{weak_this.get()}) {
+      strong_this->m_reactContext.DispatchEvent(*strong_this, L"topEnd", nullptr);
+    }
+  });
+}
+
+void ReactVideoView::OnBufferingStarted(IInspectable const &, IInspectable const &) {}
+
+void ReactVideoView::OnBufferingEnded(IInspectable const &, IInspectable const &) {}
+
+void ReactVideoView::OnSeekCompleted(IInspectable const &, IInspectable const &) {
+  runOnQueue([weak_this{get_weak()}]() {
+    if (auto strong_this{weak_this.get()}) {
+      strong_this->m_reactContext.DispatchEvent(*strong_this, L"topSeek", nullptr);
+    }
+  });
+}
+
+void ReactVideoView::Set_progressUpdateInterval(int64_t interval) {
+  m_timer.Interval(std::chrono::milliseconds{interval});
+}
+
+void ReactVideoView::Set_IsLoopingEnabled(bool value) {
+  m_isLoopingEnabled = value;
+  if (m_player != nullptr) {
+    m_player.IsLoopingEnabled(m_isLoopingEnabled);
+  }
+}
+
+void ReactVideoView::Set_UriString(hstring const &value) {
+  m_uriString = value;
+  if (m_player != nullptr) {
+    auto uri = Uri(m_uriString);
+    m_player.Source(MediaSource::CreateFromUri(uri));
+  }
+}
+
+void ReactVideoView::Set_Paused(bool value) {
+  m_isPaused = value;
+  if (m_player != nullptr) {
+    if (m_isPaused) {
+      if (IsPlaying(m_player.PlaybackSession().PlaybackState())) {
+        m_player.Pause();
+      }
+    } else {
+      if (!IsPlaying(m_player.PlaybackSession().PlaybackState())) {
+        m_player.Play();
+      }
+    }
+  }
+}
+
+void ReactVideoView::Set_Muted(bool isMuted) {
+  m_isMuted = isMuted;
+  if (m_player != nullptr) {
+    m_player.IsMuted(m_isMuted);
+  }
+}
+
+void ReactVideoView::Set_Controls(bool useControls) {
+  m_useControls = useControls;
+  AreTransportControlsEnabled(m_useControls);
+}
+
+void ReactVideoView::Set_FullScreen(bool fullScreen) {
+  if (m_fullScreen != fullScreen) {
+    m_fullScreen = fullScreen;
+    IsFullWindow(m_fullScreen);
+    auto applicationView = winrt::Windows::UI::ViewManagement::ApplicationView::GetForCurrentView();
+    bool isFullScreen = applicationView.IsFullScreenMode();
+    if (m_fullScreen) {

```

Comment:
Not a big deal to me (my apps always run in FullScreenMode) but I wonder if people expect FullWindow for the video player and FullScreen for the app to be the same thing.

One thing that could get wonky pretty fast, what if I have an app and I already entered FullScreenMode then toggle the FullScreen prop on the video. I believe that this code would cause the app to exit FullScreenMode when the video is no longer full window.
It might be better to have a separate native module that lets people set FullScreenMode independently.

Reply:
Will make fullscreen == fullwindow for now.